waku/v2/peermanager/peer_manager.go

```diff
@@ -262,7 +252,13 @@ func (pm *PeerManager) pruneInRelayConns(inRelayPeers peer.IDSlice) {
 // AddDiscoveredPeer to add dynamically discovered peers.
 // Note that these peers will not be set in service-slots.
 // TODO: It maybe good to set in service-slots based on services supported in the ENR
-func (pm *PeerManager) AddDiscoveredPeer(p PeerData) {
+func (pm *PeerManager) AddDiscoveredPeer(p PeerData, connectNow bool) {
+	//Check if the peer is already present, if so skip adding
+	_, err := pm.host.Peerstore().(wps.WakuPeerstore).Origin(p.AddrInfo.ID)
+	if err == nil || err != peerstore.ErrNotFound {
```

Comment:
```suggestion
	if err == nil || !errors.Is(err, peerstore.ErrNotFound) {
```

I'm confused with this condition. Why does an error other than `ErrNotFound` mean that the discovered peer already exist in the store?

Reply:
You are right, I thought i saw in libp2p-peerstore some weird logic of an error being returned other than errNotFound when key is present.
But on rechecking it, i am wrong.